lib/src/main/java/com/auth0/android/lock/WebProvider.java

```diff
@@ -28,15 +32,19 @@
     /**
      * Configures a new instance of the WebAuthProvider.Builder and starts it.
      *
-     * @param activity    a valid Activity context
-     * @param connection  to use in the authentication
-     * @param callback    to deliver the authentication result to
-     * @param requestCode to use in the startActivityForResult request
+     * @param activity       a valid Activity context
+     * @param connection     to use in the authentication
+     * @param authParameters extra authentication parameters to use. If null, it will use the ones defined by Options#getAuthenticationParameters
+     * @param callback       to deliver the authentication result to
+     * @param requestCode    to use in the startActivityForResult request
      */
-    public void start(Activity activity, String connection, AuthCallback callback, int requestCode) {
+    public void start(@NonNull Activity activity, @NonNull String connection, @Nullable Map<String, Object> authParameters, @NonNull AuthCallback callback, int requestCode) {
+        if (authParameters == null) {
+            authParameters = new HashMap<>(options.getAuthenticationParameters());
```

Comment:
You're right in both. Not really for the code of this method, I can use the same instance.